File: web/src/api/anniversary.ts
import { ApiResult, request } from './request';

export type AnniversaryRepeatType = 'NONE' | 'YEARLY';
export type AnniversaryDisplayType = 'COUNTDOWN' | 'PASSED' | 'TODAY';

export interface CreateAnniversaryRequest {
  relationshipId: number;
  title: string;
  description?: string;
  anniversaryDate: string;
  repeatType?: AnniversaryRepeatType;
  backgroundFileId?: number;
}

export interface UpdateAnniversaryRequest {
  title: string;
  description?: string;
  anniversaryDate: string;
  repeatType?: AnniversaryRepeatType;
  backgroundFileId?: number;
}

export interface Anniversary {
  id: number;
  relationshipId: number;
  relationshipName?: string;
  title: string;
  description?: string;
  anniversaryDate: string;
  repeatType: AnniversaryRepeatType;
  backgroundFileId?: number;
  backgroundUrl?: string;
  dayCount: number;
  displayType: AnniversaryDisplayType;
  passedYears?: number;
  createdBy: number;
  createdAt: string;
  updatedAt: string;
}

export interface GetAnniversariesParams {
  relationshipId?: number;
  repeatType?: AnniversaryRepeatType;
  displayType?: AnniversaryDisplayType;
  keyword?: string;
  page?: number;
  size?: number;
}

export function createAnniversary(data: CreateAnniversaryRequest) {
  return request.post<ApiResult<Anniversary>>('/api/anniversaries', data);
}

export function getAnniversaries(params: GetAnniversariesParams = {}) {
  return request.get<ApiResult<Anniversary[]>>('/api/anniversaries', { params });
}

export function getAnniversaryDetail(id: number) {
  return request.get<ApiResult<Anniversary>>(`/api/anniversaries/${id}`);
}

export function updateAnniversary(id: number, data: UpdateAnniversaryRequest) {
  return request.put<ApiResult<Anniversary>>(`/api/anniversaries/${id}`, data);
}

export function deleteAnniversary(id: number) {
  return request.delete<ApiResult<void>>(`/api/anniversaries/${id}`);
}

File: backend/src/main/java/com/lifelink/anniversary/entity/Anniversary.java
package com.lifelink.anniversary.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("anniversaries")
public class Anniversary {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long relationshipId;

    private String title;

    private String description;

    private LocalDate anniversaryDate;

    private String repeatType;

    private Long backgroundFileId;

    private String backgroundUrl;

    private Long createdBy;

    private Long updatedBy;

    private String status;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;
}
